File: server/datasets/java_source-code/Multiple-D1/renamed_1.java
/*
 * Quick Sort - Renamed Submission #1
 * Derived from organic_2: variables and function names renamed.
 */

class Solution {
    public static int splitHoare(int[] dataList, int startIdx, int endIdx) {
        int pivotVal = dataList[startIdx];
        int leftCursor = startIdx - 1;
        int rightCursor = endIdx + 1;
        while (true) {
            do { leftCursor++; } while (dataList[leftCursor] < pivotVal);
            do { rightCursor--; } while (dataList[rightCursor] > pivotVal);
            if (leftCursor >= rightCursor) return rightCursor;
            int swapHold = dataList[leftCursor];
            dataList[leftCursor] = dataList[rightCursor];
            dataList[rightCursor] = swapHold;
        }
    }

    public static void executeSort(int[] dataList, int startIdx, int endIdx) {
        if (startIdx < endIdx) {
            int splitPoint = splitHoare(dataList, startIdx, endIdx);
            executeSort(dataList, startIdx, splitPoint);
            executeSort(dataList, splitPoint + 1, endIdx);
        }
    }

    public static void main(String[] args) {
        int[] items = {19, 22, 63, 105, 2, 46};
        executeSort(items, 0, items.length - 1);
        System.out.println(java.util.Arrays.toString(items));
    }
}

File: server/datasets/python_source-code/Multiple-D1/renamed_1.py
"""
Quick Sort - Renamed Submission #1
Derived from organic_2: variables and function names renamed.
"""

def split_hoare(data_list, start_idx, end_idx):
    pivot_val = data_list[start_idx]
    left_cursor = start_idx - 1
    right_cursor = end_idx + 1
    while True:
        left_cursor += 1
        while data_list[left_cursor] < pivot_val:
            left_cursor += 1
        right_cursor -= 1
        while data_list[right_cursor] > pivot_val:
            right_cursor -= 1
        if left_cursor >= right_cursor:
            return right_cursor
        data_list[left_cursor], data_list[right_cursor] = data_list[right_cursor], data_list[left_cursor]

def execute_sort(data_list, start_idx, end_idx):
    if start_idx < end_idx:
        split_point = split_hoare(data_list, start_idx, end_idx)
        execute_sort(data_list, start_idx, split_point)
        execute_sort(data_list, split_point + 1, end_idx)

if __name__ == "__main__":
    items = [19, 22, 63, 105, 2, 46]
    execute_sort(items, 0, len(items) - 1)
    print(items)
